test/libsolidity/smtCheckerTests/special/tx_data_gasleft_changes.sol

```diff
@@ -0,0 +1,17 @@
+pragma experimental SMTChecker;
+
+contract C {
+	uint gleft;
+
+	function f() public payable {
+		gleft = gasleft();
+
+		fi();
+
+		assert(gleft == gasleft());
```

Comment:
How can this assertion pass if `gasleft()` itself requires some gas to execute? :)

Is this test just a stub? `gasleft` is marked as TODO in `SymbolicState` but I see some handling for it `SMTEncoder` so I'm not sure if it's unimplemented or if it is and the intention is just to move it to `SymbolicState`...

Reply:
Apparently there are some zero-cost opcodes `STOP` and `RETURN` but the abstraction at Solidity level probably won't let you just run those alone with no overhead...

I agree, depending on gas costs seems brittle. Especially that they could be completely different when the code is compiled to a different VM, like ewasm.